edisgo/network/electromobility.py

```diff
@@ -447,45 +470,312 @@ def _shorten_and_set_index(band):
                 upper_energy.loc[start + full_charging_steps, cp] += (
                     part_time_step * power
                 )
+
+        # convert to MW and cumulate energy
+        upper_power = upper_power / 1e3
+        lower_energy = lower_energy.cumsum() / hourly_steps / 1e3
+        upper_energy = upper_energy.cumsum() / hourly_steps / 1e3
+
+        # set time index
+        upper_power.index = flex_band_index
+        lower_energy.index = flex_band_index
+        upper_energy.index = flex_band_index
+
+        # write to self.flexibility_bands
+        flex_band_dict = {
+            "upper_power": upper_power,
+            "lower_energy": lower_energy,
+            "upper_energy": upper_energy,
+        }
+        self.flexibility_bands = flex_band_dict
+
+        # fix rounding errors
+        self.fix_flexibility_bands_rounding_errors(tol=tol)
+
+        edisgo_timeindex = edisgo_obj.timeseries.timeindex
+        if resample:
+            # check if time index matches Timeseries.timeindex and if not resample flex
+            # bands
+            if len(edisgo_timeindex) > 1:
+                # check if frequencies match
+                freq_edisgo = edisgo_timeindex[1] - edisgo_timeindex[0]
+                if freq_edisgo != pd.Timedelta(f"{stepsize}min"):
+                    # resample
+                    self.resample(freq=freq_edisgo)
+
         # sanity check
+        self.check_integrity()
+        # check time index
+        if len(edisgo_timeindex) > 1:
+            missing_indices = [_ for _ in edisgo_timeindex if _ not in flex_band_index]
+            if len(missing_indices) > 0:
+                logger.warning(
+                    "There are time steps in timeindex of TimeSeries object that "
+                    "are not in the index of the flexibility bands. This may lead "
+                    "to problems."
+                )
+        return self.flexibility_bands
+
+    def fix_flexibility_bands_rounding_errors(self, tol=1e-6):
```

Comment:
Should this method be tested seperately?

Reply:
I added a test for it.